File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/actions/IntakePitchUp.java
package org.firstinspires.ftc.teamcode.robot.actions;

import androidx.annotation.NonNull;

import com.acmerobotics.dashboard.telemetry.TelemetryPacket;
import com.acmerobotics.roadrunner.Action;

import org.firstinspires.ftc.teamcode.robot.Intake;

public class IntakePitchUp implements Action {
    private boolean init = false;
    private Intake i;

    @Override
    public boolean run(@NonNull TelemetryPacket packet) {
        if(!init) {
            i.pitchUp();
            init = true;
        }

        return false;
    }

    public IntakePitchUp(Intake li) {
        i = li;
    }

    public Action intakePitchUp(Intake li) {
        return new IntakePitchUp(li);
    }
}

File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/dairy/actions/IntakePitchUp.kt
package org.firstinspires.ftc.teamcode.dairy.actions

import com.acmerobotics.dashboard.telemetry.TelemetryPacket
import com.acmerobotics.roadrunner.Action
import org.firstinspires.ftc.teamcode.dairy.subsystems.IntakeClaw
import org.firstinspires.ftc.teamcode.dairy.util.Waiter

class IntakePitchUp() : Action {
    private var init = false
    private lateinit var waiter:Waiter

    override fun run(p: TelemetryPacket): Boolean {
        if (!init) {
            waiter = Waiter()
            waiter.start(800)
            IntakeClaw.pitch_pos = IntakeClaw.pitch_up
            init = true
        }

        return waiter.isDone
    }

    companion object {
        fun intakePitchUp(): Action {
            return IntakePitchUp()
        }
    }
}